src/app/controllers/locations/list.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+flnd.locationList = {
+    retrieveRecords: function(config, $scope, messenger, authService) {
+
+        $scope.list = [];
+
+        authService.get(config.locations)
+            .then(function success(response) {
+
+                $scope.list = response.data;
+
+            }, function error(response) {
+
+                messenger.displayErrorResponse($scope, response);
+
+            });
+    }
+};
+
+/**
+ * @ngdoc function
+ * @name flightNodeApp.controller:LocationListController
+ * @description
+ * # LocationListController
+ * Controller for the user list page.
+ */
+angular.module('flightNodeApp')
+    .controller('LocationListController',
+     ['$scope', '$http', '$log', 'messenger', '$location', 'authService', 'config', '$uibModal',
+        function ($scope, $http, $log, messenger, $location, authService, config, $uibModal) {
+
+            // TODO: when not authorized, an error about uiGrid will
+            // appear on the screen, probably because it tries to load
+            //  the view before changing the location path. Is there a
+            //  better place to put this? Perhaps something in the routing
+            //  to intercept the route and direct traffic by permission?
+            if (!(authService.isAdministrator() ||
+                  authService.isCoordinator())) {
+                $log.warn('not authorized to access this path');
+                $location.path('/');
+                return;
+            }
+
+            $scope.loading = true;
+
+            flnd.locationList.retrieveRecords(config, $scope, messenger, authService);
+
+            $scope.gridOptions = {
+                enableFiltering: true,
+                rowTemplate: 'app/views/row.html',
+                onRegisterApi: function (gridApi) {
+                    $scope.gridApi = gridApi;
+                },
+                data: 'list',
+                columnDefs: [
+                    { field: 'description', displayName: 'Description' },
+                    { field: 'latitude', displayName: 'Latitude' },
+                    { field: 'longitude', displayName: 'Longitude' },
+                    {
+                        field: 'id',
+                        displayName: '',
+                        cellTemplate: '\
+                        <div class="ui-grid-cell-contents" title="Edit">\
+                          <button class="btn btn-primary btn-xs" ng-click="grid.appScope.editLocation(\'{{row.entity.id}}\')" \
+                           aria-label="edit">\
+                              <span class="glyphicon glyphicon-pencil" aria-hidden="true"></span>\
+                          </button>\
+                        </div>',
+                        enableFiltering: false,
+                        width: '10%',
+                        enableColumnMenu: false
```

Comment:
Getting the setup for the grid column containing the Edit button took me a while, but now it is functioning nicely. One of the things that gave me trouble was `grid.appScope`, until I read http://ui-grid.info/docs/#/tutorial/305_appScope